### PR TITLE
feat: update version to 51.0 + jest version updates

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

 

- PR updates version metadata API version to 51.0
- Update sfdx-lwc-jest to the latest version

### What issues does this PR fix or reference?

#<Insert GitHub Issue>

## The PR fulfills these requirements:

[ ] Tests for the proposed changes have been added/updated.
[ ] Code linting and formatting was performed.

### Functionality Before

<insert gif and/or summary>

### Functionality After

<insert gif and/or summary>

### CODE REVIEW

File: sfdx-project.json [hunk 1]
```diff
@@ -3,63 +3,63 @@
...
     "packageAliases": {
         "ESObjects": "0Ho4W000000CaeJSAS",
         "ESBaseCodeLWC": "0Ho4W000000CaeTSAS",
         "ESBaseStylesLWC": "0Ho4W000000CaeOSAS",
         "ESSpaceMgmtLWC": "0Ho4W000000CaeYSAS",
-        "ESObjects@50.0.0-17": "04t4W000002PpiJQAS",
-        "ESBaseCodeLWC@50.0.0-17": "04t4W000002PpiOQAS",
-        "ESBaseStylesLWC@50.0.0-15": "04t4W000002PpiTQAS",
-        "ESSpaceMgmtLWC@50.0.0-11": "04t4W000002PpiYQAS"
+        "ESObjects@51.0.0-17": "04t4W000002PpiJQAS",
```
Comment (pozil) [inline sfdx-project.json]: You don't want to update/replace those existing package versions.

File: package.json [hunk 2]
```diff
@@ -31,8 +31,8 @@
         "@sa11y/jest": "^0.1.4-beta.0",
```
Comment (pozil) [inline package.json]: I don't think that you have the latest versions in your dev dependencies.
For comparison, here's what I have on Automation Compoents:
```
"@prettier/plugin-xml": "^0.13.0",
"@sa11y/jest": "^0.2.1",
"@salesforce/eslint-config-lwc": "^0.9.0",
"@salesforce/eslint-plugin-aura": "^1.4.0",
"@salesforce/sfdx-lwc-jest": "^0.10.4",
"eslint": "^7.19.0",
"eslint-config-prettier": "^7.2.0",
"husky": "^4.3.8",
"lint-staged": "^10.5.3",
"prettier": "^2.2.1",
"prettier-plugin-apex": "^1.8.0",
"semver": "^7.3.4"
```

**Review by pozil — CHANGES_REQUESTED** [review]

File: sfdx-project.json (hunk 1 above)
Comment (msrivastav13) [reply sfdx-project.json]: @pozil This was the alias updates! I thought with Spring 21 we want all the alias numbers to be release API version. Let me know if the build process does it.

**Review by pozil — APPROVED** [review]
All good, thanks @msrivastav13

**Pull request merged**